runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/ClueScrollPlugin.java

```diff
@@ -255,11 +262,42 @@ public void onGameTick(final GameTick event)
 
 		if (clue instanceof LocationsClueScroll)
 		{
-			final List<WorldPoint> locations = ((LocationsClueScroll) clue).getLocations();
+			final WorldPoint[] locations = ((LocationsClueScroll) clue).getLocations();
 
-			if (!locations.isEmpty())
+			if (locations.length > 0)
 			{
-				addMapPoints(locations.toArray(new WorldPoint[locations.size()]));
+				addMapPoints(locations);
+			}
+
+			if (clue instanceof ObjectClueScroll)
+			{
+				List<Integer> objectIds = ((ObjectClueScroll) clue).getObjectIds();
+
+				if (!objectIds.isEmpty()
+					&& locations.length > 0)
```

Comment:
No reason to check for ``locations.length > 0`` when you iterate over ``locations`` inside the block.